test/format/import.c

```diff
@@ -0,0 +1,215 @@
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "format.h"
+
+static void copy_file_into_directory(WT_SESSION *, const char *, const char *);
+static void get_file_metadata(WT_SESSION *, const char **, const char **);
+static void populate_table(WT_SESSION *);
+static void verify_import(WT_SESSION *);
+
+/*
+ * Import directory initialize command, remove and re-create the primary backup directory, plus a
+ * copy we maintain for recovery testing.
+ */
+#define HOME_IMPORT_INIT_CMD "rm -rf %s/" IMPORT_DIR "&& mkdir %s/" IMPORT_DIR
+#define IMPORT_DIR "IMPORT"
+/*
+ * The number of entries in the import table, primary use for validating contents after import. The
+ * varying of the entries doesn't affect functionality of import.
+ */
+#define IMPORT_ENTRIES 1000
+#define IMPORT_TABLE_CONFIG "key_format=i,value_format=i"
+#define IMPORT_URI "table:import"
+#define IMPORT_URI_FILE "file:import.wt"
+/*
+ * import --
+ *     Periodically import table.
+ */
+WT_THREAD_RET
+import(void *arg)
+{
+    WT_CONNECTION *conn, *import_conn;
+    WT_DECL_RET;
+    WT_SESSION *import_session, *session;
+    size_t cmd_len;
+    uint32_t import_value;
+    u_int period;
+    char buf[2048], cmd[64];
+    const char *file_config, *table_config;
+
+    WT_UNUSED(arg);
+    conn = g.wts_conn;
+    file_config = table_config = NULL;
+    import_value = 0;
+
+    /*
+     * Initiate a new import database, which is primarily used for copying the table file, and
+     * importing the table from a foreign database.
+     */
+    memset(cmd, 0, sizeof(cmd));
+    cmd_len = strlen(g.home) * 2 + strlen(HOME_IMPORT_INIT_CMD) + 1;
+    testutil_check(__wt_snprintf(cmd, cmd_len, HOME_IMPORT_INIT_CMD, g.home, g.home));
+    testutil_checkfmt(system(cmd), "%s", "import directory creation failed");
+
+    memset(cmd, 0, sizeof(cmd));
+    cmd_len = strlen(g.home) + strlen(IMPORT_DIR) + 10;
+    testutil_check(__wt_snprintf(cmd, cmd_len, "%s/%s", g.home, IMPORT_DIR));
+    /* Open a connection to the database, creating it if necessary. */
+    create_database(cmd, &import_conn);
+
+    /*
+     * Open two sessions, one for usual test database connection and one for the import database
+     * connection
+     */
+    testutil_check(import_conn->open_session(import_conn, NULL, NULL, &import_session));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create new table and populate with data in import database connection */
+    testutil_checkfmt(
+      import_session->create(import_session, IMPORT_URI, IMPORT_TABLE_CONFIG), "%s", g.uri);
```

Comment:
I don't think this is right. You're creating `IMPORT_URI` but if you get an error, printing out `g.uri`.